File: problems/387-first-unique-character-in-a-string/first-unique-character-in-a-string.js
/**
 * @param {string} s
 * @return {number}
 */
var firstUniqChar = function(s) {
  const freq = {};

  for (let i = 0; i < s.length; i++) {
    const c = s[i];
    if (freq[c] === undefined) freq[c] = 0;
    freq[c]++;
  }

  for (let i = 0; i < s.length; i++) {
    if (freq[s[i]] === 1) return i;
  }

  return -1;
};

File: problems/387-first-unique-character-in-a-string/first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq = {}

        for index, char in enumerate(s):
            if freq.get(char):
                freq[char][1] += 1
            else:
                freq[char] = [index, 1]

        print(freq)
        for value in freq.values():
            if value[1] == 1:
                return value[0]
        
        return -1


        